src/jobflow_remote/jobs/upgrade.py

```diff
@@ -204,6 +246,15 @@ def upgrade(
 
         versions_needing_upgrade = self.collect_upgrades(db_version, target_version)
 
+        if failed_conditions := self.check_upgrade_conditions(versions_needing_upgrade):
+            err = ["Some upgrade conditions were not satisfied:"]
+            for vv, failed_cond in failed_conditions:
+                err.append(
+                    f" - {failed_cond['condition'].description} (for version {vv}), found {failed_cond['count']}"
+                )
+            logger.error("\n".join(err))
+            return False
+
```

Comment:
It is not obvious which should be the correct behaviour, but maybe this check should also be done in the `dry_run` execution? Not as a blocking error, but as an information, that the upgrade will fail due to the current status of the DB?

Reply:
done